include/small/region.h

```diff
@@ -78,10 +79,23 @@ extern "C" {
  * region_free().
  */
 
+struct region;
+
+typedef void (*region_on_alloc)(const struct region *region,
+			        size_t size, void *opaque);
+typedef void (*region_on_truncate)(const struct region *region,
+			           size_t used, void *opaque);
+
 struct region
 {
 	struct slab_cache *cache;
 	struct slab_list slabs;
+
+	/** Callback to be called on allocation. */
+	region_on_alloc on_alloc;
+	/** Callback to be called on truncation. */
+	region_on_truncate on_truncate;
+	void *opaque;
```

Comment:
Nit: `opaque` name looks confusing, because it's unclear that it's an argument to the callbacks. Also, it lacks a comment. Maybe, rename it to `cb_arg` and rename `on_alloc`, `on_truncate` to `on_alloc_cb`, `on_truncate_cb`? This would be in agreement with our naming policy. Regarding the function types, we usually add `_f` or `_func` suffix to typedefs: `region_on_alloc_f`, `region_on_alloc_func`.

Reply:
Done.